### PR TITLE
Update sqlite-amalgamation into CMakeLists.txt

### PR DESCRIPTION
**Pull request by mosfet80**
Update sqlite-amalgamation

### CODE REVIEW

File: sqlite3_vendor/CMakeLists.txt [hunk 1]
```diff
@@ -15,8 +15,8 @@ endif()
 ament_vendor(sqlite3_vendor
   SATISFIED ${SQLite3_FOUND}
   VCS_TYPE zip
-  VCS_URL https://www.sqlite.org/2022/sqlite-amalgamation-3370200.zip
-  VCS_VERSION sqlite-amalgamation-3370200
+  VCS_URL https://www.sqlite.com/2024/sqlite-amalgamation-3450300.zip
+  VCS_VERSION ssqlite-amalgamation-3450300
```
Comment (clalancette) [inline sqlite3_vendor/CMakeLists.txt]: Changing the URL here is suspicious. This should remain sqlite.org .  Also, there is a typo in that there should only be one `s` in the VCS_VERSION.

**Review by clalancette — CHANGES_REQUESTED** [review]

File: sqlite3_vendor/CMakeLists.txt [hunk 2]
```diff
@@ -15,8 +15,8 @@ endif()
 ament_vendor(sqlite3_vendor
   SATISFIED ${SQLite3_FOUND}
   VCS_TYPE zip
-  VCS_URL https://www.sqlite.org/2022/sqlite-amalgamation-3370200.zip
-  VCS_VERSION sqlite-amalgamation-3370200
+  VCS_URL https://www.sqlite.com/2024/sqlite-amalgamation-3450300.zip
```
Comment (MichaelOrlov) [inline sqlite3_vendor/CMakeLists.txt]: As @clalancette already mentioned, the link should point to the www.sqlite.org 
```suggestion
  VCS_URL https://www.sqlite.org/2024/sqlite-amalgamation-3450300.zip
```

**Review by MichaelOrlov — CHANGES_REQUESTED** [review]

**Review by MichaelOrlov — CHANGES_REQUESTED** [review]
@mosfet80 Thank you for addressing our comments per review. LGTM.
Could you please fix DCO?
https://github.com/ros2/rosbag2/pull/1611/checks?check_run_id=24298443545

**Comment by MichaelOrlov** [comment]
@mosfet80 Friendly ping here to fix DCO in this PR.

**Comment by MichaelOrlov** [comment]
@r7vme May I ask you to pick up this PR?
This PR became stale without a response from the author for a long time. However, it would be good to have changes from this PR in our code base. I would propose creating a new PR and cherry-picking changes from this PR with mentioning co-authoring with the author of this PR and get it throw review, approval and finally merge it.

File: sqlite3_vendor/CMakeLists.txt [hunk 3]
```diff
@@ -15,8 +15,8 @@ endif()
 ament_vendor(sqlite3_vendor
   SATISFIED ${SQLite3_FOUND}
   VCS_TYPE zip
-  VCS_URL https://www.sqlite.org/2022/sqlite-amalgamation-3370200.zip
-  VCS_VERSION sqlite-amalgamation-3370200
+  VCS_URL https://www.sqlite.org/2024/sqlite-amalgamation-3450300.zip
+  VCS_VERSION sqlite-amalgamation-3450300
```
Comment (christophebedard) [inline sqlite3_vendor/CMakeLists.txt:19]: The version from Ubuntu Noble is 3.45.1: https://packages.ubuntu.com/noble/sqlite3.

Therefore, I think this should be:

```suggestion
  VCS_URL https://www.sqlite.org/2024/sqlite-amalgamation-3450100.zip
  VCS_VERSION sqlite-amalgamation-3450100
```

The `sqlite3.h` header in the ZIP file indeed shows version `3.45.1`.

Comment (christophebedard) [reply sqlite3_vendor/CMakeLists.txt:19]: I've made these changes in #1737

**Comment by MichaelOrlov** [comment]
- Closing as stale PR in favor of the #1737

**Pull request closed without merging**